examples/cache-optimization/models/member.rs

```diff
@@ -0,0 +1,111 @@
+use twilight_cache_inmemory::{model::ComputedInteractionMemberFields, CacheableMember};
+use twilight_model::{
+    application::interaction::InteractionMember,
+    gateway::payload::incoming::MemberUpdate,
+    guild::{Member, PartialMember},
+    id::{
+        marker::{RoleMarker, UserMarker},
+        Id,
+    },
+    util::{ImageHash, Timestamp},
+};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MinimalCachedMember {
+    pub user_id: Id<UserMarker>,
+    pub roles: Vec<Id<RoleMarker>>,
+    pub avatar: Option<ImageHash>,
+}
+
+impl From<Member> for MinimalCachedMember {
+    fn from(member: Member) -> Self {
+        Self {
+            user_id: member.user.id,
+            roles: member.roles,
+            avatar: member.avatar,
+        }
+    }
+}
+
+impl From<(Id<UserMarker>, PartialMember)> for MinimalCachedMember {
+    fn from((user_id, member): (Id<UserMarker>, PartialMember)) -> Self {
+        Self {
+            user_id,
+            roles: member.roles,
+            avatar: member.avatar,
+        }
+    }
+}
+
+impl
+    From<(
+        Id<UserMarker>,
+        InteractionMember,
+        ComputedInteractionMemberFields,
+    )> for MinimalCachedMember
+{
+    fn from(
+        (user_id, member, _): (
+            Id<UserMarker>,
+            InteractionMember,
+            ComputedInteractionMemberFields,
+        ),
+    ) -> Self {
+        Self {
+            user_id,
+            roles: member.roles,
+            avatar: member.avatar,
+        }
+    }
+}
+
+impl PartialEq<Member> for MinimalCachedMember {
+    fn eq(&self, other: &Member) -> bool {
+        self.user_id == other.user.id && self.roles == other.roles && self.avatar == other.avatar
+    }
+}
+
+impl PartialEq<PartialMember> for MinimalCachedMember {
+    fn eq(&self, other: &PartialMember) -> bool {
+        other
+            .user
+            .as_ref()
+            .map_or(true, |user| user.id == self.user_id)
```

Comment:
minor but members probably shouldn't be considered equal if we don't know who they are